frontend_tests/node_tests/popover_menus_data.js

```diff
@@ -0,0 +1,486 @@
+        should_display_collapse: true,
+        should_display_uncollapse: false,
+        should_display_add_reaction_option: false,
+        should_display_edit_history_option: true,
+        should_display_hide_option: false,
+        conversation_time_uri: "http://zulip.zulipdev.com/undefined/near/2",
+        narrowed: false,
+        should_display_delete_option: true,
+        should_display_read_receipts_option: true,
+        should_display_reminder_option: false,
+        should_display_quote_and_reply: true,
+    });
+});
+
+test("not_my_message_view_actions", () => {
+    set_page_params_no_edit_restrictions();
+    reset_page_params_user_status();
+    // Get message that is only viewable
+
+    const list = init_message_list();
+    message_lists.set_current(list);
+    // Message is sent by somebody else and is a stream with previous history.
+    // I should only be able to view this message with no edit/move permissions.
+    const messages = [
+        {
+            id: 1,
+            sender_id: member_user.user_id,
+            is_hidden: false,
+            sent_by_me: false,
+            locally_echoed: false,
+            is_stream: true,
+            stream_id: 1,
+            collapsed: false,
+            clean_reactions: new Map(),
+            unread: false,
+            edit_history: [
+                {
+                    prev_content: "Previous content",
+                    prev_stream: 0,
+                    prev_topic: "Previous topic",
+                },
+            ],
+        },
+    ];
+
+    add_message_with_view(list, messages);
+
+    const response = popover_menus_data.get_actions_popover_content_context(1);
+    assert.deepEqual(response, {
+        message_id: 1,
+        stream_id: 1,
+        editability_menu_item: undefined,
+        move_message_menu_item: undefined,
+        should_display_mark_as_unread: true,
+        view_source_menu_item: "translated: View message source",
+        should_display_collapse: true,
+        should_display_uncollapse: false,
+        should_display_add_reaction_option: true,
+        should_display_edit_history_option: true,
+        should_display_hide_option: false,
+        conversation_time_uri: "http://zulip.zulipdev.com/undefined/near/1",
+        narrowed: false,
+        should_display_delete_option: false,
+        should_display_read_receipts_option: true,
+        should_display_reminder_option: false,
+        should_display_quote_and_reply: true,
+    });
+});
+
+test("not_my_message_view_source_and_move", () => {
+    set_page_params_no_edit_restrictions();
+    reset_page_params_user_status();
+
+    // Get message that is movable with viewable source
+    const list = init_message_list();
+    message_lists.set_current(list);
+
+    // Message tests edge case where message is sent by someone else.
+    // Message is movable, however--I should have only view permissions with the exception of moving the message.
```

Comment:
Something like:

// Verify context for actions popover rendering for a message sent by 
// another user that the current user has permission to move.

Reply:
Sure, will make the changes.